test_util/aws.py

```diff
@@ -108,24 +109,30 @@ def delete(self):
         bucket.delete()
         logger.info('Delete successfully triggered for {}'.format(self.stack.stack_name))
 
-    def get_tag_instances(self, tag):
-        return self.boto_wrapper.client('ec2').describe_instances(
-            Filters=[
-                {'Name': 'tag-value', 'Values': [self.stack.stack_name]},
-                {'Name': 'tag-value', 'Values': [tag]}])['Reservations'][0]['Instances']
-
-    def get_master_ips(self):
-        instances = self.get_tag_instances('MasterServerGroup')
+    def get_master_ips(self, state_list=None):
+        instances = self.get_group_instances(['MasterServerGroup'], state_list)
         return instances_to_hosts(instances)
 
-    def get_public_agent_ips(self):
-        instances = self.get_tag_instances('PublicSlaveServerGroup')
+    def get_public_agent_ips(self, state_list=None):
+        instances = self.get_group_instances(['PublicSlaveServerGroup'], state_list)
         return instances_to_hosts(instances)
 
-    def get_private_agent_ips(self):
-        instances = self.get_tag_instances('SlaveServerGroup')
+    def get_private_agent_ips(self, state_list=None):
+        instances = self.get_group_instances(['SlaveServerGroup'], state_list)
         return instances_to_hosts(instances)
 
+    def get_group_instances(self, group_list, state_list=['running']):
```

Comment:
Should ideally only default state_list in the functions which will call this (get_private_agent_ips, etc). or in the function being called. It's odd to always pass it in  from all the call sites here as one value, then only in one place consistently use the "other" default (which could just be embedded in get_running_agents probably).